addons/point_of_sale/models/pos_config.py

```diff
@@ -330,6 +330,10 @@ def _check_companies(self):
         if any(self.available_pricelist_ids.mapped(lambda pl: pl.company_id.id not in (False, self.company_id.id))):
             raise ValidationError(_("The selected pricelists must belong to no company or the company of the point of sale."))
 
+    def _check_cash_control_journal(self):
+        if self.cash_control and not self.payment_method_ids.mapped('is_cash_count')[0]:
```

Comment:
it won't work because it will only check the value of the first payment method on the list [0] and will raise error if no payment method

any(pm.is_cash_count for pm in self.payment_method_ids)